### ISSUE TITLE
findAll and aggregations only work for the first 99,999 items 

### ISSUE THREAD

**Issue by UCetinkaya94** [isar/isar#184]
**Steps to Reproduce**

main.dart
```dart
import 'package:flutter/material.dart';
import 'package:isar/isar.dart';
import 'package:path_provider/path_provider.dart';

part 'main.g.dart';

@Collection()
class User {
  int? id;

  late String name;
}

late Isar isar;

void main() async {
  WidgetsFlutterBinding.ensureInitialized();

  final dir = await getApplicationSupportDirectory();

  isar = await Isar.open(
    directory: dir.path,
    schemas: [UserSchema],
  );

  runApp(const MaterialApp(
    home: HomeScreen(),
  ));
}

class HomeScreen extends StatefulWidget {
  const HomeScreen({Key? key}) : super(key: key);

  @override
  State<HomeScreen> createState() => _HomeScreenState();
}

class _HomeScreenState extends State<HomeScreen> {
  int userCount = 0;

  Future<int> get countFuture => isar.users.where().count();

  Future<List<User>> get firstUserFuture => isar.users
      .where() //
      .limit(1)
      .findAll();

  Future<List<User>> get lastUserFuture => isar.users
      .where() //
      .offset(userCount == 0 ? 0 : userCount - 1)
      .limit(1)
      .findAll();

  Future<List<User>> get findAllFuture => isar.users.where().findAll();

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      floatingActionButton: Row(
        mainAxisAlignment: MainAxisAlignment.end,
        children: [
          FloatingActionButton(
            backgroundColor: Colors.red,
            child: const Icon(Icons.delete),
            onPressed: () {
              clear();
            },
          ),
          const SizedBox(width: 12),
          FloatingActionButton(
            child: const Icon(Icons.add),
            onPressed: () {
              add25KUsers();
            },
          ),
        ],
      ),
      body: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        crossAxisAlignment: CrossAxisAlignment.stretch,
        children: [
          const Text(
            'count',
            textAlign: TextAlign.center,
            style: TextStyle(fontSize: 32),
          ),
          FutureBuilder<int>(
            future: countFuture,
            builder: (context, value) {
              if (value.hasData) {
                return Text(
                  value.data!.toString(),
                  textAlign: TextAlign.center,
                  style: const TextStyle(fontSize: 32, fontWeight: FontWeight.bold),
                );
              }

              return const SizedBox.shrink();
            },
          ),
          const SizedBox(height: 8),
          const Text(
            'findAll length',
            textAlign: TextAlign.center,
            style: TextStyle(fontSize: 32),
          ),
          FutureBuilder<List<User>>(
            future: findAllFuture,
            builder: (context, value) {
              if (value.hasData) {
                return Text(
                  value.data!.length.toString(),
                  textAlign: TextAlign.center,
                  style: const TextStyle(fontSize: 32, fontWeight: FontWeight.bold),
                );
              }

              return const SizedBox.shrink();
            },
          ),
          const SizedBox(height: 8),
          const Text(
            'first user',
            textAlign: TextAlign.center,
            style: TextStyle(fontSize: 32),
          ),
          FutureBuilder<List<User>>(
            future: firstUserFuture,
            builder: (context, value) {
              if (value.hasData) {
                final name = value.data!.isNotEmpty ? value.data!.first.name : '-';

                return Text(
                  name,
                  textAlign: TextAlign.center,
                  style: const TextStyle(fontSize: 32, fontWeight: FontWeight.bold),
                );
              }

              return const SizedBox.shrink();
            },
          ),
          const SizedBox(height: 8),
          const Text(
            'last user',
            textAlign: TextAlign.center,
            style: TextStyle(fontSize: 32),
          ),
          FutureBuilder<List<User>>(
            future: lastUserFuture,
            builder: (context, value) {
              if (value.hasData) {
                final name = value.data!.isNotEmpty ? value.data!.first.name : '-';

                return Text(
                  name,
                  textAlign: TextAlign.center,
                  style: const TextStyle(fontSize: 32, fontWeight: FontWeight.bold),
                );
              }

              return const SizedBox.shrink();
            },
          ),
        ],
      ),
    );
  }

  Future<void> add25KUsers() async {
    final users = [
      for (int id = userCount; id < userCount + 25000; id++) //
        User()
          ..id = id
          ..name = 'User ${id + 1}',
    ];

    await isar.writeTxn((isar) async {
      await isar.users.putAll(users);
    });

    setState(() {
      userCount += 25000;
    });
  }

  Future<void> clear() async {
    await isar.writeTxn((isar) async {
      await isar.users.deleteAll([
        for (int id = 0; id < userCount; id++) //
          id
      ]);
    });

    setState(() {
      userCount = 0;
    });
  }
}

```

https://user-images.githubusercontent.com/67158575/149399850-983cc84b-50f9-41b7-b245-5bd286e70d7a.mov

**Version**
 - Platform: only tested on the iOS Simulator (iOS 15.2)
 - Flutter version: 2.8.1
 - Isar version: 1.0.5
 - Isar Generator version: 1.0.5

**Comment by leisim** [comment]
Thanks for the report! Will be fixed in the next release.

**Comment by leisim** [comment]
Fixed in `2.0.0-dev.0`

**Comment by UCetinkaya94** [comment]
> Fixed in `2.0.0-dev.0`

You're a legend ❤️

**Comment by ayalma** [comment]
Hi body
When do you going to release 2.0.0 stable version?
